### PR TITLE
Intellisence comments for lease acquiring functions.

### PR DESCRIPTION
**Pull request by Udith-Gayan**

### CODE REVIEW

File: src/clients/tenant-client.js [hunk 1]
```diff
@@ -19,6 +19,12 @@ const TenantEvents = {
 
 class TenantClient extends BaseEvernodeClient {
 
+    /**
+     *
```
Comment (chalith) [inline src/clients/tenant-client.js:23]: Constructs a tenant client instance.

File: src/clients/tenant-client.js [hunk 2]
```diff
@@ -19,6 +19,12 @@ const TenantEvents = {
 
 class TenantClient extends BaseEvernodeClient {
 
+    /**
+     * 
+     * @param {string} xrpAddress The address of the XRPL account of the tenant.
```
Comment (chalith) [inline src/clients/tenant-client.js:24]: XRPL address of the tenant.
XRPL secret of the tenant.

File: src/clients/tenant-client.js [hunk 3]
```diff
@@ -56,6 +62,13 @@ class TenantClient extends BaseEvernodeClient {
         return host;
     }
 
+    /**
+     * 
+     * @param {string} hostAddress he address of the XRPL account of the host.
```
Comment (chalith) [inline src/clients/tenant-client.js:67]: XRPL address of the host to acquire the lease.

File: src/clients/tenant-client.js [hunk 4]
```diff
@@ -56,6 +62,13 @@ class TenantClient extends BaseEvernodeClient {
...
+     * @param {object} requirement The instance configuration.
```
Comment (chalith) [inline src/clients/tenant-client.js:68]: requirements

File: src/clients/tenant-client.js [hunk 5]
```diff
@@ -56,6 +62,13 @@ class TenantClient extends BaseEvernodeClient {
...
+     * @param {object} options The transaction configuration.
```
Comment (chalith) [inline src/clients/tenant-client.js:69]: options for the XRPL transaction.

File: src/clients/tenant-client.js [hunk 6]
```diff
@@ -83,6 +96,12 @@ class TenantClient extends BaseEvernodeClient {
         return this.xrplAcc.buyNft(selectedOfferIndex, [{ type: MemoTypes.ACQUIRE_LEASE, format: MemoFormats.BASE64, data: ecrypted }], options.transactionOptions);
     }
 
+    /**
+     * Watch for the acquire-success response after the acquiring request made.
+     * @param {object} tx The transaction resolved by the acquireLeaseSubmit function.
```
Comment (chalith) [inline src/clients/tenant-client.js:101]: The transaction returned bu..

File: src/clients/tenant-client.js [hunk 7]
```diff
@@ -83,6 +96,12 @@ class TenantClient extends BaseEvernodeClient {
         return this.xrplAcc.buyNft(selectedOfferIndex, [{ type: MemoTypes.ACQUIRE_LEASE, format: MemoFormats.BASE64, data: ecrypted }], options.transactionOptions);
     }
 
+    /**
+     * Watch for the acquire-success response after the acquiring request made.
```
Comment (chalith) [inline src/clients/tenant-client.js:100]: after the acquire request is made

File: src/clients/tenant-client.js [hunk 8]
```diff
@@ -83,6 +96,12 @@ class TenantClient extends BaseEvernodeClient {
...
+     * @param {object} tx The transaction resolved by the acquireLeaseSubmit function.
+     * @param {object} options The transaction configuration passed to the acquireLease function.
```
Comment (chalith) [inline src/clients/tenant-client.js:102]: Options for the XRPL transaction.

File: src/clients/tenant-client.js [hunk 9]
```diff
@@ -83,6 +96,12 @@ class TenantClient extends BaseEvernodeClient {
...
+     * @returns A promise that resolves with an object including transaction details,instance info, and acquireReference Id.
```
Comment (chalith) [inline src/clients/tenant-client.js:103]: A object including.. No need to mention about promise

File: src/clients/tenant-client.js [hunk 10]
```diff
@@ -121,6 +140,13 @@ class TenantClient extends BaseEvernodeClient {
             }
     }
 
+    /**
+     * Acquire an instance from a host
+     * @param {string} hostAddress The address of the XRPL account of the host.
```
Comment (chalith) [inline src/clients/tenant-client.js:145]: XRPL address of the host to acquire the lease.

File: src/clients/tenant-client.js [hunk 11]
```diff
@@ -121,6 +140,13 @@ class TenantClient extends BaseEvernodeClient {
...
+     * @param {object} requirement The instance configuration.
```
Comment (chalith) [inline src/clients/tenant-client.js:146]: requirements

File: src/clients/tenant-client.js [hunk 12]
```diff
@@ -121,6 +140,13 @@ class TenantClient extends BaseEvernodeClient {
...
+     * @param {object} options The transaction configuration.
```
Comment (chalith) [inline src/clients/tenant-client.js:147]: Options for the XRPL transaction

File: src/clients/tenant-client.js [hunk 13]
```diff
@@ -121,6 +140,13 @@ class TenantClient extends BaseEvernodeClient {
...
+     * @returns A promise that resolves with an object including transaction details,instance info, and acquireReference Id.
```
Comment (chalith) [inline src/clients/tenant-client.js:148]: A object including.. No need to mention about promise

**Review by kasunkavinda — APPROVED** [review]

**Pull request merged**